clients/drcachesim/tools/histogram.cpp

```diff
@@ -95,19 +95,39 @@ histogram_t::parallel_shard_exit(void *shard_data)
     return true;
 }
 
+static inline addr_t
+back_align(addr_t addr, addr_t align)
+{
+    return addr & ~(align - 1);
+}
+
 bool
 histogram_t::parallel_shard_memref(void *shard_data, const memref_t &memref)
 {
     shard_data_t *shard = reinterpret_cast<shard_data_t *>(shard_data);
+    std::unordered_map<addr_t, uint64_t> *cache_map = nullptr;
+    addr_t start_addr;
+    size_t size;
     if (type_is_instr(memref.instr.type) ||
-        memref.instr.type == TRACE_TYPE_PREFETCH_INSTR)
-        ++shard->icache_map[memref.instr.addr >> line_size_bits_];
-    else if (memref.data.type == TRACE_TYPE_READ ||
-             memref.data.type == TRACE_TYPE_WRITE ||
-             // We may potentially handle prefetches differently.
-             // TRACE_TYPE_PREFETCH_INSTR is handled above.
-             type_is_prefetch(memref.data.type))
-        ++shard->dcache_map[memref.data.addr >> line_size_bits_];
+        memref.instr.type == TRACE_TYPE_PREFETCH_INSTR) {
```

Comment:
curious as why we didn't use `type_is_prefetch(memref.instr.type)` instead here?

Reply:
That would return true for data prefetches also.  We only want to detect instruction prefetches here.